rplidar.go

```diff
@@ -399,13 +405,13 @@ func (d *Device) getPointCloud(ctx context.Context) (pointcloud.PointCloud, time
 		d.Start()
 	}
 
-	// discard scans for warmup
+	// wait and then discard scans for warmup
 	if !d.scannedOnce {
 		d.scannedOnce = true
+		utils.SelectContextOrWait(ctx, time.Duration(d.warmupNumDiscardedScans)*time.Second)
```

Comment:
I don't think you want time.Duration() on this, assuming one second per discarded scan, just  `d.warmupNumDiscardedScans*time.Second`

time.Duration() actually represents nanoseconds, and multiplying nanoseconds by seconds would make no sense here (output units would be like seconds^2 or something)

Reply:
Great catch! I actually have to multiply a `time.Duration` with a `time.Duration` unless I have a static variable I'm using to multiply `time.Second` with.

I solved it by multiplying what I pass into `time.Duration` with `math.Pow(10,9)`.

Reply:
oh - nvm - the code does what I initially guessed it would do, because:

```golang
time.Millisecond =  1ms
time.Second =  1s
time.Minute =  1m0s

time.Duration(1) = 1ns
time.Duration(1) * time.Millisecond =  1ms
time.Duration(1) * time.Second =  1s
time.Duration(1) * time.Minute =  1m0s
```

Reply:
Yeah, I caught this because it was logically nonsensical to multiple durations, and with static numbers you can just multiply directly, but apparently what you're doing is correct/required for Go semantics.